fuse/data/datasets/dataset_default.py

```diff
@@ -187,8 +187,9 @@ def getitem(
         # get sample id
         if not self._explicit_sample_ids_mode:
             sample_id = item
-            if sample_id >= self._final_sample_ids:
-                raise IndexError
+            if isinstance(sample_id, (int, np.integer)):  # allow using non int sample_ids
```

Comment:
So it means that works?
If yes, for the official fix - I want to support it more explicitly.
In the constructor, 
We will support setting sample_ids to None.
If it will set to None - it means that Dataset has no control over the used sample ids, and therefore iter() and len() will raise an error.

Reply:
yes it worked. 
I made some changes, let me know if it's better